lib/msal-node/src/client/IConfidentialClientApplication.ts

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { AuthenticationResult, AuthorizationCodeRequest, AuthorizationUrlRequest, ClientCredentialRequest, Logger, OnBehalfOfRequest, RefreshTokenRequest, SilentFlowRequest } from "@azure/msal-common";
+import { NodeConfigurationAuthError } from "../error/NodeConfigurationAuthError";
+import { TokenCache } from "../cache/TokenCache";
+export interface IConfidentialClientApplication {
+    getAuthCodeUrl(request: AuthorizationUrlRequest): Promise<string>;
+    acquireTokenByCode(request: AuthorizationCodeRequest): Promise<AuthenticationResult>;
+    acquireTokenSilent(request: SilentFlowRequest): Promise<AuthenticationResult>;
+    acquireTokenByRefreshToken(request: RefreshTokenRequest): Promise<AuthenticationResult>;
+    acquireTokenByClientCredential(request: ClientCredentialRequest): Promise<AuthenticationResult>;
+    acquireTokenOnBehalfOf(request: OnBehalfOfRequest): Promise<AuthenticationResult>;
+    getTokenCache(): TokenCache;
+    getLogger(): Logger;
+    setLogger(logger: Logger): void;
+}
+
+export const stubbedConfidentialClientApplication: IConfidentialClientApplication = {
```

Comment:
What's the use case for this in Node?

Reply:
This may not be needed for `node`, there is no need to stub client application instances as you pointed out. Will remove this.

Reply:
Done